sqladmin/models.py

```diff
@@ -283,6 +288,88 @@ class UserAdmin(ModelAdmin, model=User):
     edit_template: ClassVar[str] = "edit.html"
     """Edit view template. Default is `edit.html`."""
 
+    # Form
+    form: ClassVar[Optional[Type[Form]]] = None
+    """Form class.
+    Override if you want to use custom form for your model.
+    Will completely disable form scaffolding functionality.
+
+    ???+ example
+        ```python
+        class MyForm(Form):
+            name = StringField('Name')
+
+        class MyModelAdmin(ModelAdmin, model=User):
+            form = MyForm
+        ```
+    """
+
+    form_base_class: ClassVar[Optional[Type[Form]]] = Form
+    """Base form class.
+    Will be used by form scaffolding function when creating model form.
+    Useful if you want to have custom constructor or override some fields.
+
+    ???+ example
+        ```python
+        class MyBaseForm(Form):
+            def do_something(self):
+                pass
+
+        class MyModelAdmin(ModelAdmin, model=User):
+            form_base_class = MyBaseForm
+        ```
+    """
+
+    form_args: ClassVar[Optional[Dict[str, Dict[str, Any]]]] = None
```

Comment:
And is Optional required here? Can we default to `{}` as this is not a function argument?

Reply:
We can avoid the `Optional` here, yes.